torchvision/datasets/celeba.py

```diff
@@ -53,7 +53,8 @@ def __init__(self, root,
                  transform=None, target_transform=None,
                  download=False):
         import pandas
-        super(CelebA, self).__init__(root)
+        root = os.path.join(root, self.base_folder)
+        super(CelebA, self).__init__(root, root_zipfilename=os.path.join(root, "img_align_celeba.zip"))
```

Comment:
I don't quite like that this happens implicitly.

Do you think it would be possible to just let the user pass `root` as the the zip filename, and in `VisionDataset` identify that `root` is a zip, and perform all that is needed?

Reply:
For e.g. the Omniglot dataset, the ZIP filename depends on the subset of the dataset to use (background=True/False => images_background.zip/images_evaluation.zip). So there passing the name of the ZIP file would break the abstraction. I want to keep the original ZIP files of the dataset as much as possible.

Reply:
hum, ok, let me think a bit more about this.